src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -1,15 +1,246 @@
+    | Error err ->
+        return (Error err)
+
+  (* run Coda CLI command *)
+  let run_coda ~logger ~name ~namespace ~cmd ~subcmd ~flags =
+    if Option.is_none !full_name_opt then
+      failwith "run_coda: node name has not been set" ;
+    let kubectl_args =
+      ["exec"; name; "--namespace"; namespace; "-c"; "coda"; "--"]
+    in
+    let coda_args = ["coda"; cmd; subcmd] @ flags in
+    let args = kubectl_args @ coda_args in
+    [%log info] "Running kubectl %s\n" String.(concat args ~sep:" ") ;
+    Process.run_lines_exn ~prog:"kubectl" ~args ()
+
   let start _ = failwith "TODO"
 
   let stop _ = failwith "TODO"
 
-  let send_payment _ _ = failwith "TODO"
+  module Decoders = Graphql_lib.Decoders
+
+  module Graphql = struct
+    let port = 3085
+
+    (* queries on localhost because of port forwarding *)
+    let uri =
+      Uri.make
+        ~host:Unix.Inet_addr.(localhost |> to_string)
+        ~port:3085 ~path:"graphql" ()
+
+    let set_port_forwarding ~logger ~name ~namespace =
+      let args =
+        ["port-forward"; name; "--namespace"; namespace; string_of_int port]
+      in
+      [%log info] "Running kubectl %s\n" String.(concat args ~sep:" ") ;
+      let%bind.Deferred.Or_error.Let_syntax proc =
+        Process.create ~prog:"kubectl" ~args ()
+      in
+      Exit_handlers.register_handler ~logger
+        ~description:"Kubectl port forwarder" (fun () ->
+          ignore Signal.(send kill (`Pid (Process.pid proc))) ) ;
+      Process.collect_stdout_and_wait proc
+
+    module Client = Graphql_lib.Client.Make (struct
+      let preprocess_variables_string = Fn.id
+
+      let headers = String.Map.empty
+    end)
+
+    module Unlock_account =
+    [%graphql
+    {|
+          mutation ($password: String!,
+          $public_key: PublicKey!) {
+             unlockAccount(input: {password: $password, publicKey: $public_key }) {
+                 public_key: publicKey @bsDecoder(fn: "Decoders.public_key")
+             }
+          }
+    |}]
+
+    module Send_payment =
+    [%graphql
+    {|
+          mutation ($sender: PublicKey!,
+          $receiver: PublicKey!,
+          $amount: UInt64!,
+          $token: UInt64,
+          $fee: UInt64!,
+          $nonce: UInt32,
+          $memo: String) {
+          sendPayment(input:
+            {from: $sender, to: $receiver, amount: $amount, token: $token, fee: $fee, nonce: $nonce, memo: $memo}) {
+              payment {
+        id
+      }
+    }
+  }
+  |}]
+  end
+
+  let send_payment ~logger ~namespace ~node ~sender ~receiver ~amount ~fee =
+    [%log info] "Running send_payment test"
+      ~metadata:[("namespace", `String namespace); ("node", `String node)] ;
+    let open Deferred.Or_error.Let_syntax in
+    let%bind () = set_name_from_network ~logger ~namespace ~node in
+    let name = Option.value_exn (get_full_name ()) in
+    Deferred.don't_wait_for
+      ( match%map.Deferred.Let_syntax
+          Graphql.set_port_forwarding ~logger ~name ~namespace
```

Comment:
Will this setup port forwarding every time we send a gql request? It might make more sense to just setup the gql port forwarding for all nodes during the deployment step. Then you could store the port on the `Node.t` type directly.

Reply:
Well, it will start forwarding each time this test is run.

We can start port forwarding with a distinct port on localhost for each block-producer node.

Reply:
I'm adding a mechanism to allow port forwarding from a specified port. For now, I'm going to leave the call in the test startup, but it will be straightforward to move it to `deploy` or another location.